qiskit/providers/ibmq/jupyter/dashboard/watcher_monitor.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A module of widgets for job monitoring"""
+import sys
+import time
+import threading
+from ...utils.converters import utc_to_local
+
+
+def _job_monitor(job, status, watcher):
+    """Monitor the status of a IBMQJob instance.
+
+    Args:
+        job (BaseJob): Job to monitor.
+        status (Enum): Job status.
+        watcher (JobWatcher): Job watcher instance
+    """
+    thread = threading.Thread(target=_job_checker, args=(job, status, watcher))
+    thread.start()
+
+
+def _job_checker(job, status, watcher):
+    """A simple job status checker
+
+    Args:
+        job (BaseJob): The job to check.
+        status (Enum): Job status.
+        watcher (JobWatcher): Job watcher instance
+
+    """
+    prev_status_name = None
+    prev_queue_pos = None
+    interval = 2
+    exception_count = 0
+
+    while status.name not in ['DONE', 'CANCELLED', 'ERROR']:
+        time.sleep(interval)
+        try:
+            status = job.status()
+            exception_count = 0
+
+            if status.name == 'QUEUED':
+                queue_pos = job.queue_position()
+                if queue_pos != prev_queue_pos:
+                    est_time = utc_to_local(job.queue_info().estimated_start_time)
+
+                    update_info = (job.job_id(), status.name+' ({})'.format(queue_pos),
+                                   est_time.strftime("%H:%M %Z (%m/%d)"), status.value)
```

Comment:
Doesn't `update_single_job` expects the 3rd element to be the queue position?

Reply:
Nope, but that should be clarified in the docs now.  The queue position is just part of the string in arg[1].